tests/uv/mlflow/test_reporter.py

```diff
@@ -201,3 +204,76 @@ def test_report():
         cur_step = s['step']
         for k, v in s['m'].items():
           assert metric_data[k][cur_step] == v
+
+
+def test_pubsub_report(monkeypatch):
+
+  with tempfile.TemporaryDirectory() as tmpdir:
+    mlf.set_tracking_uri(f'file:{tmpdir}/foo')
+    _reset_experiment()
+
+    # we need to access the run via a client to inspect most params/tags/metrics
+    client = mlf.tracking.MlflowClient()
+
+    class MockClient():
+
+      def __init__(self, batch_settings=(), publisher_options=(), **kwargs):
+        pass
+
+      def publish(self, topic: str, msg: bytes):
```

Comment:
can you use the actual implementation that the consumer will use?

Reply:
think we need to mock pubsub here for testing, right?